### ISSUE TITLE
Horovod & Tensorflow, how to get the average evaluate_accuracy?  Estimator.evaluate on multiple(N) GPUs, each GPU uses 1/n data.

### ISSUE THREAD

**Issue by zhangyanzi01** [horovod/horovod#2073]
**Environment:**
1. Framework: TensorFlow
2. Framework version: 2.0.0
3. Horovod version:0.19.0
4. MPI version:v3.1
5. CUDA version:cuda10.0
6. NCCL version:nccl_2.4.7
7. Python version:3.6.6
8. OS and version:RHEL7.4 
9. GCC version: gcc version 4.8.5 20150623 (Red Hat 4.8.5-16) (GCC)

**Your question:**

Scenario:
I use Horovod & Tensorflow, not use keras.
1. Use "tf.estimator.Estimator.train" for training on N GPUs, each GPU trains 1/N training data;
2. After several epochs, do evaluate;
3.Use "tf.estimator.Estimator.evaluate" to evaluate, and each GPU evaluates 1/N evaluation data

My question:
Each GPU will have an valuation_result,

**Comment by zhangyanzi01** [comment]
My question:
Each GPU will have an evaluation_result, 
How to get the average evaluation_result among all GPUs evaluation_results?

**Comment by zhangyanzi01** [comment]
Waiting for your reply, thank you!

**Comment by tgaddair** [comment]
Hey @zhangyanzi01, there are a couple options.  If you're using MPI, you can use `mpi4py.allreduce` to sum the value across workers, then divide by `hvd.size()` to get the average.

Alternatively, you can use `tf.convert_to_tensor()` on the evaluation accuracy, then run `hvd.allreduce(op=hvd.Average)` to get the average.  Both should work.
